tests/test_helpers.py

```diff
@@ -66,3 +70,280 @@ def test_local_model_non_default_name(self, tmp_path):
         # with adapter name
         result = check_if_peft_model(tmp_path / "peft-gpt2-other" / "other")
         assert result is True
+
+
+class TestScalingAdapters:
+    def get_scale_from_modules(self, model):
+        layer_to_scale_map = {}
+        for name, module in model.named_modules():
+            if isinstance(module, LoraLayer):
+                layer_to_scale_map[name] = module.scaling
+
+        return layer_to_scale_map
+
+    def test_set_adapter_scale(self):
+        model = AutoModelForCausalLM.from_pretrained("facebook/opt-125m")
+        lora_config = LoraConfig(
+            r=4,
+            lora_alpha=4,
+            target_modules=["k_proj", "v_proj"],
+            lora_dropout=0.1,
+            bias="none",
+            init_lora_weights=False,
+        )
+
+        model = get_peft_model(model, lora_config)
+        model.eval()
+        tokenizer = AutoTokenizer.from_pretrained("facebook/opt-125m")
+        inputs = tokenizer("hello world", return_tensors="pt")
+
+        with torch.no_grad():
+            logits_before_scaling = model(
+                **inputs,
+            ).logits
+
+        scales_before_scaling = self.get_scale_from_modules(model)
+
+        with set_adapter_scale(model=model, alpha=0.5):
+            scales_during_scaling = self.get_scale_from_modules(model)
+            for key in scales_before_scaling.keys():
+                assert scales_before_scaling[key] != scales_during_scaling[key]
+
+            with torch.no_grad():
+                logits_during_scaling = model(**inputs).logits
+
+            assert not torch.allclose(logits_before_scaling, logits_during_scaling)
+
+        scales_after_scaling = self.get_scale_from_modules(model)
+        for key in scales_before_scaling.keys():
+            assert scales_before_scaling[key] == scales_after_scaling[key]
+
+        with torch.no_grad():
+            logits_after_scaling = model(**inputs).logits
+
+        assert torch.allclose(logits_before_scaling, logits_after_scaling)
+
+    def test_wrong_scaling_datatype(self):
```

Comment:
Could this test be simplified to just check for the error? All the stuff with the logits doesn't really add much, right?

Reply:
Ah! That makes sense to me.